ui/src/features/Editor/components/OverlayUI/components/NodePickerDialog/index.tsx

```diff
@@ -64,6 +65,7 @@ const NodePickerDialog: React.FC<Props> = ({
           />
           <Select
             value={currentActionByType}
+            disabled={currentActionByType === "transformer" && !isMainWorkflow}
```

Comment:
I think the second suggestion is best and more logically accurate. We don't want things disabled if the currentActionByType is transformer and then is a subworkflow. We want it if the action is writer or reader and in a subworkflow